ops/pebble.py

```diff
@@ -3211,6 +3314,34 @@ def get_notices(
         resp = self._request('GET', '/v1/notices', query)
         return [Notice.from_dict(info) for info in resp['result']]
 
+    def get_identities(self) -> Dict[str, Identity]:
+        """Get all identities in Pebble.
+
+        Returns:
+            A dict mapping identity names to :class:`Identity` objects.
+        """
+        resp = self._request('GET', '/v1/identities')
+        result = resp['result']
+        return {name: Identity.from_dict(d) for name, d in result.items()}
+
+    def replace_identities(
+        self, identities: Mapping[str, Union[IdentityDict, Identity, None]]
+    ) -> None:
+        """Replace the named identities in Pebble with the given ones.
+
```

Comment:
This "don't have a remove method" decision seems like it's "because we don't have to have one". Is there more to it than that? I don't like that the Pebble CLI is more user-friendly than the Python client. It seems like it would be trivial to have a `remove_identities` method that just called replaced under the hood with appropriate None values, and that would be user-friendly and not seem like exposing an implementational detail.

Reply:
Yeah, that's fair. It's not hard to add `remove_identities` (though I'd have it call the "remove" Pebble API rather than going via replace).

@tonyandrewmeyer Do you think we should also add `add_identities` and `update_identities` too, for completeness? I think those are almost always better done with `replace_identities`, especially in a charm context, where you want idempotency.

Reply:
I wouldn't object to `add` and `update` but I don't feel those are as necessary. A single method for add/update seems clean and not-common, and indeed there's some idempotency benefits. It was specifically remove having the special "put in None" that felt like bad DX to me.

Reply:
Per discussion in the daily, we decided to add `remove_identities` with the parameter being a set of strings (names of the identities to be removed). I will implement this and add some unit tests and adjust the real pebble test accordingly.